examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/ConnectionExampleFragment.java

```diff
@@ -152,9 +228,124 @@ public void handle(MatterError err) {
             });
   }
 
+  private void displayPasscodeInputDialog(Context context) {
+    AlertDialog.Builder builder = new AlertDialog.Builder(context);
+
+    LayoutInflater inflater = LayoutInflater.from(context);
+    View dialogView = inflater.inflate(R.layout.custom_passcode_dialog, null);
+
+    // Set up the input dialog with the default passcode
+    final EditText input = dialogView.findViewById(R.id.passcode_input);
+    input.setText(DEFAULT_COMMISSIONER_GENERATED_PASSCODE_STRING);
+
+    // Set up the buttons
+    builder.setPositiveButton(
+        "Continue Connecting",
+        new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) {
+            String passcode = input.getText().toString();
+            Log.i(
+                TAG,
+                "displayPasscodeInputDialog() User entered CastingPlayer/Commissioner-Generated passcode: "
+                    + passcode);
+
+            // Display the user entered passcode on the screen
+            connectionFragmentStatusTextView.setText(
+                "Continue Connecting with user entered CastingPlayer/Commissioner-Generated passcode: "
+                    + passcode
+                    + "\n\n");
+
+            long passcodeLongValue = DEFAULT_COMMISSIONER_GENERATED_PASSCODE_INT;
+            try {
+              passcodeLongValue = Long.parseLong(passcode);
+              Log.i(
+                  TAG,
+                  "displayPasscodeInputDialog() User entered CastingPlayer/Commissioner-Generated passcode: "
+                      + passcodeLongValue);
+            } catch (NumberFormatException nfe) {
+              Log.e(
+                  TAG,
+                  "displayPasscodeInputDialog()User entered CastingPlayer/Commissioner-Generated passcode is not a valid integer. NumberFormatException: "
+                      + nfe);
+              connectionFragmentStatusTextView.setText(
+                  "User entered CastingPlayer/Commissioner-Generated passcode is not a valid integer: "
+                      + passcode
+                      + "\n\n");
+            }
+
+            // Update the CommissionableData DataProvider and AndroidChipPlatform with the user
```

Comment:
AndroidChipPlatform is hidden from the client developer now. Avoid mentioning that in the documentation.

```suggestion
            // Update the CommissionableData DataProvider with the user
```